Automattic/wp-calypso
# Sidebar: Add expand and collapse actions

#### Changes proposed in this Pull Request

* Refactors my-sites sidebar state to use a keyed reducer for sidebar sections
* Adds expand and collapse actions for sidebar sections

#### Testing instructions

* Load My Sites section of Calypso
* Expand and collapse each section of the sidebar (Site, Design, Tools, Manage)
* With at least 1 sidebar section expanded, reload the page--the section should be expanded once the page has reloaded (if state is hydrated on reload)

Supports #33519
